sig/datadog/core/configuration/agent_settings_resolver.rbs

```diff
@@ -2,45 +2,74 @@ module Datadog
   module Core
     module Configuration
       class AgentSettingsResolver
-        AgentSettings: untyped
+        class AgentSettings < ::Struct[untyped]
```

Comment:
I will leave that exercise for me on a future PR :)

Reply:
Note: as we add more things to `sig/datadog/{core,tracing,...}`, we may want to add some `check ...` in `Steepfile` so that their corresponding source also gets checked.